packages/sanity/src/core/releases/hooks/useDocumentVersions.tsx

```diff
@@ -33,30 +59,27 @@ const INITIAL_VALUE = {
  */
 export function useDocumentVersions(props: DocumentPerspectiveProps): DocumentPerspectiveState {
   const {documentId} = props
-
   const publishedId = getPublishedId(documentId)
-
   const documentPreviewStore = useDocumentPreviewStore()
+  const [results, setResults] = useState<DocumentPerspectiveState>(INITIAL_VALUE)
 
   const observable = useMemo(() => {
-    return documentPreviewStore
-      .unstable_observeDocumentIdSet(`sanity::versionOf("${publishedId}")`, undefined, {
-        apiVersion: RELEASES_STUDIO_CLIENT_OPTIONS.apiVersion,
-      })
-      .pipe(
-        swr(`${publishedId}`),
-        map(({value}) => ({
-          data: value.documentIds,
-          loading: false,
-          error: null,
-        })),
-        catchError((error) => {
-          return of({error, data: [] as string[], loading: false})
-        }),
-      )
+    const cachedObservable = observableCache.get(publishedId)
+    if (cachedObservable) {
+      return cachedObservable
+    }
+
+    const newObservable = createObservable(documentPreviewStore, publishedId)
+    observableCache.set(publishedId, newObservable)
```

Comment:
just struck me that this will cause problems if the same document id is used in different workspaces. The cache key should include projectId+dataset as well

Reply:
Oooh fantastic catch let me fix that

Reply:
Done :)